### PR TITLE
8267305: regenerate clang jextract binding

### PR DESCRIPTION
**Pull request by sundararajana**
clang.symbols file to list the symbols used.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Change must be properly reviewed

### Issue
 * [JDK-8267305](https://bugs.openjdk.java.net/browse/JDK-8267305): regenerate clang jextract binding

### Reviewers
 * [Maurizio Cimadamore](https://openjdk.java.net/census#mcimadamore) (@mcimadamore - Committer)
 * [Jorn Vernee](https://openjdk.java.net/census#jvernee) (@JornVernee - Committer) ⚠️ Review applies to f2ef97cf7164e2ce29c21a5eb0bb1ee533b524ab

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/panama-foreign pull/538/head:pull/538` \
`$ git checkout pull/538`

Update a local copy of the PR: \
`$ git checkout pull/538` \
`$ git pull https://git.openjdk.java.net/panama-foreign pull/538/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 538`

View PR using the GUI difftool: \
`$ git pr show -t 538`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/panama-foreign/pull/538.diff">https://git.openjdk.java.net/panama-foreign/pull/538.diff</a>

</details>

### CODE REVIEW

File: src/jdk.incubator.jextract/share/classes/jdk/internal/clang/Index.java [hunk 1]
```diff
@@ -83,13 +84,14 @@ public ParsingFailedException(Path srcFile, ErrorCode code) {
 
     public TranslationUnit parseTU(String file, Consumer<Diagnostic> dh, int options, String... args)
     throws ParsingFailedException {
-        try (NativeScope scope = NativeScope.unboundedScope()) {
+        try (ResourceScope scope = ResourceScope.newConfinedScope()) {
```
Comment (mcimadamore) [inline src/jdk.incubator.jextract/share/classes/jdk/internal/clang/Index.java:87]: Watch out for indentation in this method - e.g. the `throws` clause seems odd on the new line. Also, there are missing spaces before/after the conditional operator `?`

File: src/jdk.incubator.jextract/share/classes/jdk/internal/clang/TranslationUnit.java [hunk 2]
```diff
@@ -81,17 +82,18 @@ void processDiagnostics(Consumer<Diagnostic> dh) {
         }
     }
 
-    static long FILENAME_OFFSET = Index_h.CXUnsavedFile.$LAYOUT().bitOffset(MemoryLayout.PathElement.groupElement("Filename")) / 8;
-    static long CONTENTS_OFFSET = Index_h.CXUnsavedFile.$LAYOUT().bitOffset(MemoryLayout.PathElement.groupElement("Contents")) / 8;
-    static long LENGTH_OFFSET = Index_h.CXUnsavedFile.$LAYOUT().bitOffset(MemoryLayout.PathElement.groupElement("Length")) / 8;
+    static long FILENAME_OFFSET = CXUnsavedFile.$LAYOUT().bitOffset(MemoryLayout.PathElement.groupElement("Filename")) / 8;
```
Comment (mcimadamore) [inline src/jdk.incubator.jextract/share/classes/jdk/internal/clang/TranslationUnit.java:85]: can we use byteOffset instead of bitOffset / 8

**Review by mcimadamore — COMMENTED** [review]
Looks really good and much more compact than what we had before!
I left some minor code comments on code not autogenerated.

**Review by JornVernee — APPROVED** [review]
LGTM

**Review by mcimadamore — APPROVED** [review]
Looks good

**Pull request closed without merging**